uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractTestBroker.java

```diff
@@ -0,0 +1,129 @@
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.broker;
+
+import android.text.TextUtils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.BuildConfig;
+import com.microsoft.identity.client.ui.automation.app.App;
+import com.microsoft.identity.client.ui.automation.installer.LocalApkInstaller;
+import com.microsoft.identity.client.ui.automation.installer.PlayStore;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+import static com.microsoft.identity.client.ui.automation.utils.CommonUtils.FIND_UI_ELEMENT_TIMEOUT;
+import static com.microsoft.identity.client.ui.automation.utils.CommonUtils.getResourceId;
+
+/**
+ * A model for interacting with a Broker App during UI Test.
+ */
+public abstract class AbstractTestBroker extends App implements ITestBroker {
+
+    public AbstractTestBroker(@NonNull final String packageName,
+                              @NonNull final String appName) {
+        super(packageName, appName, BuildConfig.INSTALL_SOURCE_LOCAL_APK
+                .equalsIgnoreCase(BuildConfig.BROKER_INSTALL_SOURCE)
+                ? new LocalApkInstaller() : new PlayStore());
+    }
+
+    @Override
+    public void handleAccountPicker(@Nullable final String username) {
+        final UiDevice device = UiDevice.getInstance(getInstrumentation());
+
+        // find the object associated to this username in account picker.
+        // if the username is not provided, then click on the "Use another account" option
+        final UiObject accountSelected = device.findObject(new UiSelector().resourceId(
+                getResourceId(getPackageName(), "account_chooser_listView")
+        ).childSelector(new UiSelector().textContains(
+                // This String is pulled from
+                // R.string.broker_account_chooser_choose_another_account
+                TextUtils.isEmpty(username) ? "Use another account" : username
+        )));
+
+        try {
+            accountSelected.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+            accountSelected.click();
+        } catch (final UiObjectNotFoundException e) {
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    @Override
+    public void performJoinViaJoinActivity(@NonNull final String username,
```

Comment:
The ultimate goal is to publish this library/module i.e `uiautomationutilities` as an artifact in ADO. Then we can start maintain different versions of this library and allow us to locally change from `local` to `dist`. It would then make sense for anyone working on broker UI to also make changes to UI Test utilities for broker that listed in this class and a few other classes. This would allow us to debug/run the tests with local UI changes and also makes sure that UI automation utilities are updated accordingly without breaking existing tests.